src/box/tuple.h

```diff
@@ -170,6 +170,26 @@ box_tuple_format(box_tuple_t *tuple);
 const char *
 box_tuple_field(box_tuple_t *tuple, uint32_t fieldno);
 
+/**
+ * Return the raw tuple field pointed by a full JSON path in
+ * MsgPack format.
+ *
+ * The JSON path includes the outmost field. For example, "c" in
+ * ["a", ["b", "c"], "d"] can be accessed using '[2][2]' path. The
+ * array element accessors are one-based (as in Lua).
```

Comment:
This is inconsistent with `box_tuple_field`, which uses 0-based indexes. I think we should either use 0-based indexes, like `box_tuple_field`, or allow to pass `index_base`, as in `box_update`.

Reply:
I like the idea, thanks! Will update it.

Reply:
Added the `index_base` argument.

I need it in several existing functions, added in a preparatory commit. Expaded testing to catch incorrect interpretatoin of the array element selector for outmost/nested field and in case of using an offset map.